src/mlpack/tests/perceptron_test.cpp

```diff
@@ -23,6 +24,61 @@ using namespace mlpack::distribution;
 BOOST_AUTO_TEST_SUITE(PerceptronTest);
 
 /**
+ * This test tests whether the SimpleWeightUpdate updates weigths and biases correctly.
+ */
+BOOST_AUTO_TEST_CASE(SimpleWeightUpdateUpdateWeights)
+{
+  SimpleWeightUpdate wip;
+
+  vec trainingPoint("1 2 3 4 5");
+  mat weights("0 1 6;"
+              "2 3 6;"
+              "4 5 6;"
+              "6 7 6;"
+              "8 9 6");
+  vec biases("2 5 7");
+  size_t incorrectClass = 0;
+  size_t correcClass = 2;
+
+  wip.UpdateWeights(trainingPoint, weights, biases, incorrectClass, correcClass);
+
+  BOOST_CHECK_EQUAL(weights(0, 0), -1);
+  BOOST_CHECK_EQUAL(weights(1, 0), 0);
+  BOOST_CHECK_EQUAL(weights(2, 0), 1);
+  BOOST_CHECK_EQUAL(weights(3, 0), 2);
+  BOOST_CHECK_EQUAL(weights(4, 0), 3);
+
+  BOOST_CHECK_EQUAL(weights(0, 2), 7);
+  BOOST_CHECK_EQUAL(weights(1, 2), 8);
+  BOOST_CHECK_EQUAL(weights(2, 2), 9);
+  BOOST_CHECK_EQUAL(weights(3, 2), 10);
+  BOOST_CHECK_EQUAL(weights(4, 2), 11);
```

Comment:
Ack, sorry, ignore this.  I forgot that SimpleWeightUpdate only updates one positive and negative class at a time.  (I wonder if that should be updated, since there can be more than one incorrect class.)